packages/compass-sidebar/src/components/multiple-connections/connections-navigation.tsx

```diff
@@ -123,10 +109,10 @@ type ConnectionListTitleActions =
 type ConnectionsNavigationComponentProps = {
   connectionsWithStatus: ReturnType<typeof useConnectionsWithStatus>;
   activeWorkspace: WorkspaceTab | null;
-  filterRegex: RegExp | null;
-  excludeInactive: boolean;
-  onFilterChange(regex: RegExp | null): void;
-  onToggleExcludeInactive(): void;
+  filter: ConnectionsFilter;
```

Comment:
I've combined the `filterRegex` and `excludeInactive` as adding the pop-over suggests we might be adding more parameters for filtering in the future.

Reply:
Makes sense. I believe that's the plan with the popover.